diff --git a/bin/Flingbox.apk b/bin/Flingbox.apk
index e13827d..15b2b22 100644
Binary files a/bin/Flingbox.apk and b/bin/Flingbox.apk differ
diff --git a/bin/classes.dex b/bin/classes.dex
index 440f597..b46816b 100644
Binary files a/bin/classes.dex and b/bin/classes.dex differ
diff --git a/bin/edu/eside/flingbox/ObjectSettingsDialog.class b/bin/edu/eside/flingbox/ObjectSettingsDialog.class
index 9cd8e0d..dad07ff 100644
Binary files a/bin/edu/eside/flingbox/ObjectSettingsDialog.class and b/bin/edu/eside/flingbox/ObjectSettingsDialog.class differ
diff --git a/bin/edu/eside/flingbox/R$id.class b/bin/edu/eside/flingbox/R$id.class
index 1fd87bf..b7ffd39 100644
Binary files a/bin/edu/eside/flingbox/R$id.class and b/bin/edu/eside/flingbox/R$id.class differ
diff --git a/bin/edu/eside/flingbox/R$layout.class b/bin/edu/eside/flingbox/R$layout.class
index 2219e71..496f68b 100644
Binary files a/bin/edu/eside/flingbox/R$layout.class and b/bin/edu/eside/flingbox/R$layout.class differ
diff --git a/bin/edu/eside/flingbox/R$string.class b/bin/edu/eside/flingbox/R$string.class
index b4dc01a..6568d3e 100644
Binary files a/bin/edu/eside/flingbox/R$string.class and b/bin/edu/eside/flingbox/R$string.class differ
diff --git a/bin/edu/eside/flingbox/math/Intersect.class b/bin/edu/eside/flingbox/math/Intersect.class
index 88c63df..b647a8f 100644
Binary files a/bin/edu/eside/flingbox/math/Intersect.class and b/bin/edu/eside/flingbox/math/Intersect.class differ
diff --git a/bin/edu/eside/flingbox/physics/PhysicBody.class b/bin/edu/eside/flingbox/physics/PhysicBody.class
index 096f256..0c1e1aa 100644
Binary files a/bin/edu/eside/flingbox/physics/PhysicBody.class and b/bin/edu/eside/flingbox/physics/PhysicBody.class differ
diff --git a/bin/resources.ap_ b/bin/resources.ap_
index d6753a0..bb2bcad 100644
Binary files a/bin/resources.ap_ and b/bin/resources.ap_ differ
diff --git a/gen/edu/eside/flingbox/R.java b/gen/edu/eside/flingbox/R.java
index 293d7d2..89cbcc2 100644
--- a/gen/edu/eside/flingbox/R.java
+++ b/gen/edu/eside/flingbox/R.java
@@ -16,7 +16,9 @@ public final class R {
         public static final int text_stone=0x7f020002;
     }
     public static final class id {
-        public static final int settings_list=0x7f050000;
+        public static final int checkbox_lock_body=0x7f050000;
+        public static final int seek_bodys_density=0x7f050001;
+        public static final int settings_list=0x7f050002;
     }
     public static final class layout {
         public static final int body_settings=0x7f030000;
diff --git a/res/layout/body_settings.xml b/res/layout/body_settings.xml
index 9805358..8fccede 100644
--- a/res/layout/body_settings.xml
+++ b/res/layout/body_settings.xml
@@ -1,7 +1,35 @@
 <?xml version="1.0" encoding="utf-8"?>
 
-<ListView
-  xmlns:android="http://schemas.android.com/apk/res/android"
-  android:id="@+id/settings_list"
-  android:layout_width="wrap_content"
-  android:layout_height="wrap_content"/>
+<TableLayout xmlns:android="http://schemas.android.com/apk/res/android" 
+	android:orientation="vertical" 
+	android:stretchColumns="0" 
+	android:layout_gravity="center_vertical|center_horizontal" 
+ 	android:layout_height="wrap_content" 
+	android:layout_width="wrap_content"> 
+
+	<!-- Lock Body -->
+	<TableRow> 
+		<TextView android:textColor="#ffffff" 
+			android:text="@string/body_settings_lock_body" 
+			android:textSize="16px" 
+			android:layout_marginLeft="6dip"/> 
+		<CheckBox android:id="@+id/checkbox_lock_body" 
+			android:layout_gravity="right|top" /> 
+	</TableRow> 
+     
+     <!-- Density -->
+	<TableRow> 
+		<TextView android:textColor="#ffffff" 
+			android:text="@string/body_settings_density" 
+			android:textSize="16px" 
+			android:layout_marginLeft="6dip"/>     
+	</TableRow> 
+	<SeekBar android:id="@+id/seek_bodys_density" 
+			android:layout_width="fill_parent" 
+			android:layout_height="wrap_content"
+			android:layout_gravity="right|top" 
+			android:layout_marginLeft="6dip"
+			android:layout_marginRight="6dip"/>     
+
+
+</TableLayout> 
diff --git a/src/edu/eside/flingbox/ObjectSettingsDialog.java b/src/edu/eside/flingbox/ObjectSettingsDialog.java
index 23b08c5..f0b2b0e 100644
--- a/src/edu/eside/flingbox/ObjectSettingsDialog.java
+++ b/src/edu/eside/flingbox/ObjectSettingsDialog.java
@@ -18,15 +18,22 @@
 
 package edu.eside.flingbox;
 
+import edu.eside.flingbox.objects.AtomicBody;
 import android.app.Dialog;
 import android.content.Context;
+import android.content.DialogInterface;
 import android.os.Bundle;
 import android.view.View;
 import android.view.View.OnClickListener;
+import android.widget.CheckBox;
 
 public class ObjectSettingsDialog extends Dialog implements OnClickListener {
 
-	Context mContext;
+	private Context mContext;
+	
+	protected AtomicBody mBody;
+	
+	private CheckBox mLocked;
 
 	public ObjectSettingsDialog(Context context) {
 		super(context);
@@ -43,12 +50,14 @@ public class ObjectSettingsDialog extends Dialog implements OnClickListener {
 
     	// Sets dialog title
 		setTitle(R.string.body_settings_name);
-
+		
+		mLocked = (CheckBox)findViewById(R.id.checkbox_lock_body);
+		mLocked.setOnClickListener(this);
     }
 
 	@Override
 	public void onClick(View v) {
-		
+
 		
 	}
 
diff --git a/src/edu/eside/flingbox/math/Intersect.java b/src/edu/eside/flingbox/math/Intersect.java
index 8fb5fce..182ddff 100644
--- a/src/edu/eside/flingbox/math/Intersect.java
+++ b/src/edu/eside/flingbox/math/Intersect.java
@@ -60,21 +60,6 @@ public class Intersect {
 	}
 	
 	public static Intersect[] intersectPolygons(Vector2D[] polygonA, Vector2D[] polygonB) {
-		/* find smaller polygon */
-		Vector2D[] smallerPolygon, largerPolygon;
-		if (polygonA.length > polygonB.length) {
-			smallerPolygon = polygonA;
-			largerPolygon = polygonB;
-		} else {
-			smallerPolygon = polygonB;
-			largerPolygon = polygonA;
-		}
-		
-		/* Check if first point is inside the other polygon */
-		int firstSegment = 0, lastSegment = smallerPolygon.length - 1;
-		
-		if (PolygonUtils.polygonConatinsPoint(polygonB, polygonA[0])) {
-		}
 		
 		return null;
 	}
diff --git a/src/edu/eside/flingbox/physics/PhysicBody.java b/src/edu/eside/flingbox/physics/PhysicBody.java
index e08edc0..a21570b 100755
--- a/src/edu/eside/flingbox/physics/PhysicBody.java
+++ b/src/edu/eside/flingbox/physics/PhysicBody.java
@@ -59,9 +59,9 @@ public abstract class PhysicBody {
 	/** body's mass by square unit */
 	protected float mDensity;
 	/** friction to be applied when body is moving */
-	protected float mDinamicFrictionCoeficient = 0.25f;
+	protected float mDinamicFrictionCoeficient = 0.5f;
 	/** friction to be applied when body isn's moving */
-	protected float mStaticFrictionCoeficient = 0.33f;
+	protected float mStaticFrictionCoeficient = 0.67f;
 	
 	/** Object's current position */
 	protected final Vector2D mPosition;
